### ISSUE TITLE
Catastrophic cluster failure on v20160916.0

### ISSUE THREAD

**Issue by temujin9** [flynn/flynn#3424]
At 11:23 local time, our qa cluster melted down more or less completely, and never recovered.

Collecting debug info results in multiple ```EROR[09-24|22:15:34] error getting scheduler state err="object_not_found: no leader found"``` results.

This appears to have coincided with the ZFS pool reaching it's boundary (90% of disk, per an earlier recommendation). It may therefore be a duplicate of other bugs (in particular the "no image caching" and "no image garbage collection" issues), but its notable that it completely killed this cluster and resulted in a 6hr hard (user-facing) outage due to a coinciding issue (#3423) with standing up new clusters.

**Comment by temujin9** [comment]
`collect-debug-info` hung on the third node, at the `getting scheduler state` step. Another half-gig tarball coming your way: https://bazaarvoice.box.com/s/jspheexzv05klr1m2gkhq16lp135z449

**Comment by lmars** [comment]
Sorry it has took a while to get back to you on this one.

Here is a description of what seems to have happened:

Discoverd on `ip-10-100-37-16` was the leader at `16:17 UTC` and begun marking all instances for expiry over a period of 85s from `16:17:58` to `16:19:23`. Here is the breakdown per 10s period:

```
$ grep '2016-09-24T16.*marking instance for expiry' 5a54f058-5082-4206-acbc-fce888ad144e.log | grep -oP '2016-09-24T16:\d\d:\d' | uniq -c
   8 2016-09-24T16:17:5
  40 2016-09-24T16:18:0
  48 2016-09-24T16:18:1
  10 2016-09-24T16:18:2
  18 2016-09-24T16:18:3
   6 2016-09-24T16:18:4
   6 2016-09-24T16:19:0
   2 2016-09-24T16:19:1
   2 2016-09-24T16:19:2
```

So note that it wasn't a case of all services going down at once.

This seems to be an issue on the server side because there are no heartbeat errors in any of the logs in that same period, so my suspicion is that heartbeats were being received but the full disk somehow blocked the server from being able to process them.

@temujin9 do you know which of the hosts had a full disk? We can probably try and reproduce this by filling up a ZFS pool and seeing what happens.

/cc @josephglanville

**Comment by lmars** [comment]
@temujin9 by the way, we fixed `flynn-host collect-debug-info` blocking on collecting scheduler state in #3476.

**Comment by temujin9** [comment]
Honestly, the catastrophic issue I've been having recently (which I tracked down to the AWS CLI, not your code) have completely eclipsed my memory of this incident. I do not remember which host was full, and it would be a lot of difficult work to determine at this remove.

I haven't had it reproduce lately, but I've also expanded all my disks in size, and been cycling every cluster fairly regularly.

Thanks for fixing the `collect-debug-info` again.
